File: src/structural/flyweight/typescript/classes/DeliveryLocation.ts
import { DeliveryFlyweight } from './interfaces/DeliveryFlyweight';
import { DeliveryLocationData } from './interfaces/DeliveryLocationData';

export class DeliveryLocation implements DeliveryFlyweight {
  constructor(private _intrinsicState: DeliveryLocationData) {}

  get intrinsicState(): DeliveryLocationData {
    return this._intrinsicState;
  }

  deliver(name: string, number: string): void {
    console.log(`Deliver to ${name}`);
    console.log(
      `On street ${this._intrinsicState.street},`,
      `city ${this._intrinsicState.city}`,
    );
    console.log(`Number: ${number}`);
    console.log(`---------------------------`);
  }
}

File: src/structural/flyweight/python/classes/delivery_location.py
from classes.interfaces.delivery_flyweight_protocol import \
    DeliveryFlyweightProtocol
from classes.interfaces.delivery_location_data import DeliveryLocationData


class DeliveryLocation(DeliveryFlyweightProtocol):
    def __init__(self, intrinsic_state: DeliveryLocationData) -> None:
        self.__INTRINSIC_STATE = intrinsic_state

    @property
    def intrinsic_state(self) -> DeliveryLocationData:
        return self.__INTRINSIC_STATE

    def deliver(self, name: str, number: str) -> None:
        print(f'Deliver to {name}')
        print(
            f'On street {self.intrinsic_state.street}' +
            f', city {self.intrinsic_state.city}')
        print(f'Number: {number}')
        print('----------------------------')

    def __repr__(self) -> str:
        return str(f'{{ {self.intrinsic_state} }}')
